x-pack/plugins/canvas/canvas_plugin_src/renderers/dropdown_filter/component/dropdown_filter.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiIcon } from '@elastic/eui';
+import PropTypes from 'prop-types';
+import React, { ChangeEvent, FocusEvent, SFC } from 'react';
+
+export interface Props {
+  /**
+   * A collection of choices to display in the dropdown
+   * @default []
+   */
+  choices?: string[];
+  /** Optional value for the component */
+  value?: string;
+  /** Function to invoke when the dropdown value is changed */
+  onChange: (value: string) => void;
+  /** Function to invoke when the dropdown value is committed */
+  commit: (value: string) => void;
+}
+
+export const DropdownFilter: SFC<Props> = ({ value, onChange, commit, choices = [] }) => {
+  choices = choices || [];
+
+  // Prepend the choices with the value provided if it isn't in the choices array.
+  if (value && !choices.includes(value)) {
+    choices = [value].concat(choices);
+  }
```

Comment:
Does it make sense to add a new value to the list of options? We will populate the dropdown with the unique values found the column selected, so there shouldn't be any need to enter your own value, and there currently isn't a way, besides editing the HTML directly, to pass in your value. If you do enter your own value, you won't get any matching data because that value shouldn't exist in the filtered column.  Instead if we get a new value, I think it'd make sense to check if the value is in our list of options and to maybe default to commit '%%CANVAS_MATCH_ALL%%', so it doesn't blow away all your data on your workpad.

If we keep this logic, can we make an exception for `%%CANVAS_MATCH_ALL%%`? It's getting prepended as a separate option, but it's a duplicate of the first.

![feb-22-2019 17-12-03](https://user-images.githubusercontent.com/1697105/53278227-470cf700-36c5-11e9-9ba0-372b4ee26635.gif)

Reply:
I'll just remove the adding in favor of just discarding, and document it.  I was approaching this holistically, but the use case makes sense.